common/src/main/java/com/sequenceiq/cloudbreak/auth/altus/Crn.java

```diff
@@ -591,32 +600,45 @@ public static Builder builder() {
 
         private String resource;
 
-        public Builder setPartition(Partition partition) {
-            this.partition = checkNotNull(partition);
+        /**
+         * @deprecated ALTUS was replaced by CDP and is kept here for backward compatibility reasons (e.g., dynamodb serialized CRNs).
+         */
+        @Deprecated
+        public Builder setOldPartition() {
+            this.partition = Partition.ALTUS;
             return this;
         }
 
-        public Builder setService(Service service) {
-            this.service = checkNotNull(service);
+        public Builder setAccountId(String accountId) {
+            this.accountId = checkNotNull(accountId);
             return this;
         }
 
-        public Builder setAccountId(String accountId) {
-            this.accountId = checkNotNull(accountId);
+        public Builder setResource(String resource) {
+            this.resource = checkNotNull(resource);
             return this;
         }
 
-        public Builder setResourceType(ResourceType resourceType) {
-            this.resourceType = checkNotNull(resourceType);
+        public Crn build(CrnResourceDescriptor resourceDescriptor) {
```

Comment:
is the resourceDescriptor required or why do you add to the param list?

Reply:
yes, required

Reply:
As I know, almost all of the fields are required in the crn. Is it a good pattern to add the required fields as a parameter to the build() method?

Reply:
all fields are required, but some of them have 1 value always, some of them changed to package private in this change and alo crnresourcedescriptor is not part of the crn, so it is not a required field, but helps to fill in some fields of the crn, so it i decided to add to build() as parameter

Reply:
i can also add a new constructor for builder if it is better, instead of adding a parameter to build(), what do you think @topolyai5 ?